### PR TITLE
Move to a primary mixin

### PR DESCRIPTION
**Pull request by rowanmanning**
This adds in a primary mixin. A lot of mixins have been made private,
most of them won't cause an issue as they're only used in one
decomissioned service.

Other places that the public mixins were used should be relatively
easy to refactor - the only reason the mixins are being used is to
have custom class names. From what I can tell in n-messaging-client
this has lead to lots of repeated styles output anyway.

Apologies for the seemingly large set of changes – a lot of them are
me adding in more full JSDoc.

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 2.7.11 to 44a28b0a9b46bcb19199875152a272a38318a894
No bundle size difference found.

File: src/scss/_brand.scss [hunk 1]
```diff
@@ -69,6 +80,10 @@
 			shaded-close-text: oColorsGetPaletteColor('white'),
 			shaded-close-border: oColorsGetPaletteColor('white')
 		),
-		'supports-variants': ()
+		'supports-variants': (
+			'compact',
```
Comment (notlee) [inline src/scss/_brand.scss]: As all brands support all variants, is this and `_oOverlaySupports('...')` needed?

File: src/scss/overlay.scss [hunk 2]
```diff
@@ -31,25 +53,34 @@
...
 	overflow: auto;
 }
 
-/// Base overlay content style
-@mixin oOverlayContent {
+/// Outputs styles for the overlay body.
+///
+/// @output The output does not include a class definition, and should be wrapped in a selector.
+/// @access private
+@mixin oOverlayContentBody {
```
Comment (notlee) [inline src/scss/overlay.scss]: This is marked `@access private` but the name doesn't reflect that

**Comment by rowanmanning** [comment]
Good spots @notlee, fixed both

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 2.7.11 to 193f080c3194e60adc44b2a6362d50eb878057db
No bundle size difference found.

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 2.7.11 to 1c75c3e9f1fc86cb4ca436d5babd229743d00cd2
No bundle size difference found.

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 2.7.11 to e1f4dbef5fe9f88c1aac17bf00dff1618c8dee2d
No bundle size difference found.

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 2.7.11 to 1397f3d1a597591e97a740a898c18995098f33c5
No bundle size difference found.

**Pull request merged**